src/sql/migrations.rs

```diff
@@ -501,6 +501,40 @@ item TEXT DEFAULT '');"#,
         sql.execute_migration("ALTER TABLE msgs ADD COLUMN hop_info TEXT;", 81)
             .await?;
     }
+    if dbversion < 82 {
+        info!(context, "[migration] v82");
+        sql.execute_migration(
+            r#"CREATE TABLE imap (
+id INTEGER PRIMARY KEY AUTOINCREMENT,
+rfc724_mid TEXT DFEAULT '', -- Message-ID header
+folder TEXT DEFAULT '', -- IMAP folder
+target TEXT DEFAULT '', -- Destination folder, empty to delete.
+uid INTEGER DEFAULT 0, -- UID
+uidvalidity INTEGER DEFAULT 0,
+UNIQUE (folder, uid, uidvalidity)
+);
+CREATE INDEX imap_folder ON imap(folder);
+CREATE INDEX imap_messageid ON imap(rfc724_mid);
+
+INSERT INTO imap
+(rfc724_mid, folder, target, uid, uidvalidity)
+SELECT
+rfc724_mid,
+server_folder AS folder,
+server_folder AS target,
```

Comment:
No, it's not guaranteed. The resulting effect is that all pending move jobs are lost and some messages may remain in the inbox if they were about to be moved when you stopped the app to upgrade. But otherwise the migration will be too complex (it already is), I am trying to keep it as simple as possible so it doesn't fail.

Reply:
i assume whatever the lost jobs were trying to do would effectively be retried after the migration which is what makes this fine?

Reply:
No, if the message was in the inbox, and there was a pending job to move it to the movebox, the job will become invalid (`job::load_next` will return an error and the job will be thrown away), and migration will set `target` to "Inbox", so the message will never be moved. I am just not sure it's worth the effort to try to migrate the move jobs and turn them into `target="DeltaChat"` in the IMAP table. It requires a `JOIN` of `msgs` table with jobs table, it's hard to test handling all the corner cases when there are multiple messages with the same UID in `msgs` table, and it's unlikely that you have a move job in the jobs table during migration. Worst case is that a few messages remain in the Inbox: this already happens sometimes anyway (#2163) and will not happen after merging this PR or will be easy to debug.